pkg/policy/mapstate.go

```diff
@@ -788,19 +788,16 @@ func (e *mapStateEntry) merge(entry *mapStateEntry) {
 			}
 		}
 
-		// Explicit auth takes precedence over defaulted one.
-		if entry.hasAuthType == ExplicitAuthType {
-			if e.hasAuthType == ExplicitAuthType {
-				// Numerically higher AuthType takes precedence when both are explicitly defined
-				if entry.AuthType > e.AuthType {
-					e.AuthType = entry.AuthType
-				}
-			} else {
-				e.hasAuthType = ExplicitAuthType
+		// Numerically higher AuthType takes precedence when both are
+		// either explicitly defined or derived
+		if entry.hasAuthType == e.hasAuthType {
```

Comment:
can this be replaced with a `max()`? or do we need to track non-set as well?

Reply:
We could use `max()` as well for setting the `entry.AuthType` to the higher of the two.